### ISSUE TITLE
ScatterPlotItem doesn't update on setBrush, setSize, setPen, setSymbol when pxMode=False

### ISSUE THREAD

**Issue by pierstitus** [pyqtgraph/pyqtgraph#915]
When an ScatterPlotItem is updated with setBrush, setSize, setPen or setSymbol while pxMode=False, the screen is not updated. With pxMode=True it does work.

A workaround is to call .invalidate() afterwards.
setData also does update, but requires you to set all data, as it clears all points and adds new points internally. (which could be clarified in the setData documentation, btw)

Following example shows the problem.
```
from pyqtgraph.Qt import QtGui, QtCore
import pyqtgraph as pg
import numpy as np

win = pg.GraphicsWindow()

pos = np.random.rand(10,2)
size = 0.3 * np.random.rand(10)
brush = np.random.rand(10)

cmap = pg.ColorMap([0,1],[[0,0,0,255],[255,255,255,255]])

plot = win.addPlot()
plotitem = pg.ScatterPlotItem(pos=pos, size=size, brush=cmap.mapToQColor(brush), pxMode=False)
plot.addItem(plotitem)

plot2 = win.addPlot()
plotitem2 = pg.ScatterPlotItem(pos=pos, size=size, brush=cmap.mapToQColor(brush), pxMode=True)
plot2.addItem(plotitem2)

def update():
    global pos, size, brush
    size = 0.3 * np.random.rand(10)
    brush = np.random.rand(10)

    plotitem.setBrush(cmap.mapToQColor(brush))
    plotitem.setSize(size)

    # calling invalidate does the update:
    # plotitem.invalidate()

    # this also works:
    # plotitem.setData(pos=pos, size=size, brush=cmap.mapToQColor(brush))

    plotitem2.setBrush(cmap.mapToQColor(brush))
    plotitem2.setSize(100*size)

timer = QtCore.QTimer()
timer.timeout.connect(update)
timer.start(200)

## Start Qt event loop unless running in interactive mode.
if __name__ == '__main__':
    import sys
    if (sys.flags.interactive != 1) or not hasattr(QtCore, 'PYQT_VERSION'):
        QtGui.QApplication.instance().exec_()
```

Also the setBrush, setSize, setPen and setSymbol don't emit a sigPlotChanged signal, which I would expect, at least when they set an array (different for each point).

**Comment by ixjlyons** [comment]
Thanks for the report and the good example. I can verify it doesn't update and I think it should (`invalidate` is not set to `True` [here](https://github.com/pyqtgraph/pyqtgraph/blob/172120106d2baaaff454be6a200e07069644ac6c/pyqtgraph/graphicsItems/ScatterPlotItem.py#L553) when not in `pxMode`). It seems like a deliberate choice so I will investigate further.

Not emitting `sigPlotChanged` for setting brush/pen/etc. may have been deliberate as well and would be a change that users may already be working around, so perhaps we could add a new signal that's emitted when there's a stylistic change.

I agree that it should be clearly pointed out that `setData` clears *everything*.

**Comment by pierstitus** [comment]
To be more specific, `setData` clears all point data, the default properties are kept.

When a plot property changes that feels more like a stylistic change, while changing per point properties that feels more like a data change to me.

Thanks for the quick reaction!

**Comment by timjolson** [comment]
I suggest adding a `setOptions` or `setStyle` that can accept and update any style options, and emits either the sigPlotChanged or a new signal (that can be used to push legend updates/plot redraw).

**Comment by NilsNemitz** [comment]
Did this issue get closed by #1420?

**Comment by j9ac9k** [comment]
Can confirm this is now fixed, likely from #1420
